proposals/0001-unify-nat-and-natural.md

```diff
@@ -0,0 +1,71 @@
+---
+author: Rinat Stryungis
+date-accepted: ""
+ticket-url: "https://gitlab.haskell.org/ghc/ghc/-/issues/10776"
+implemented: "Yes"
+---
+
+This proposal is [discussed at this pull request](https://github.com/haskell-core/core-libraries-proposals/pull/0>).
+**After creating the pull request, edit this file again, update the number in
+the link, and delete this bold sentence.**
+
+# Proposal title
+
+The changes in the proposal remove the separate built-in kind `Nat` 
+in favor of promoted type `Natural`.
+
+
+## Motivation
+
+Existence of separated kind `Nat` and impossibility of promotion 
+of usual `Natural` data type (and data constructors with `Natual` fields)
+looks like an unnecessary and inconvenient complication. 
+It forces to create redundant types with `Nat` fields in their constructors (it makes these types uninhabited in terms)
+only for promotional purposes. This is more cumbersome and less safe than the promotion 
+of the ordinary data type. 
+
+## Proposed Change Specification
+
+1. This patch removes the separate kind `Nat` and repaces it by promoted type `Natural`. 
```

Comment:
```suggestion
1. Remove the separate kind `Nat` and replace it with `Natural`. Type-level numeric literals now have the kind `Natural`, i.e. `p :: Proxy (1 :: Nat)` becomes `p :: Proxy (1 :: Natural)`.
```

Reply:
The patch should be mentioned in the implementation section, but not here. This section should just describe the change itself, so let's avoid phrases such as "This patch removes ..." here.

Also, let's make it clear that the typing rules for type-level literals have changed, i.e. `1 :: Nat` becomes `1 :: Natural`.